geps/gep-1619.md

```diff
@@ -130,98 +275,563 @@ Generally, the implementation API programs the dataplane API; however these two
+for example use cases on traffic splitting.
+
+### Cookie Attributes
+
+While the API is intended to be generic, as described in [API Granularity](#api-granularity), a majority of
+implementations will employ session persistence through cookies. Therefore, let's explore the possibilities of cookie
+configuration for these APIs.
+
+A cookie is composed of various attributes, each represented as key=value pairs. While some attributes may have optional
+values, the cookie name attribute is the only mandatory one, and the rest are considered optional.
+
+The cookie attributes defined by [RFC6265](https://www.rfc-editor.org/rfc/rfc6265#section-5.2) are:
+- Name=_value_
 - Expires=_date_
 - Max-Age=_number_
 - Domain=_domain_
 - Path=_path-value_
 - Secure
 - HttpOnly
 
-However, there are de-facto standard attributes that have been proposed as updates to [rfc6265](https://www.rfc-editor.org/rfc/rfc6265#section-5.2):
-- SameSite=_[Strict|Lax|None]_ ([Draft RFC Update](https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-cookie-same-site-00))
-- Partitioned ([Draft RFC Update](https://www.ietf.org/archive/id/draft-cutler-httpbis-partitioned-cookies-01.html))
-
-**Session Cookie Strategy**
-
-Session cookie strategy is configuration specific to the creation of the cookie by the dataplane. These are not specific configuration parameters of the cookie itself, but instead the operation to take when inserting the cookie such as:
-- Rewriting an existing cookie value
-- Inserting a new cookie
-- Prefixing an existing cookie
-- Preserving an existing cookie value
-- Only adding cookies on specific HTTP methods (POST/GET)
+Other cookie attributes not defined by RFC6265, but are captured in draft RFCs and could be considered de facto
+standards due to wide acceptance are:
+- SameSite=[Strict|Lax|None]
+- Partitioned
 
-### Session Affinity API
-
-TBD - See [open question](#open-questions) about session affinity API
+All cookie attributes are considered as opaque values in this spec and are to be determined by the individual
+implementations. To be precise, given the `Name` cookie attribute is opaque, clients should not expect a deterministic
+name. However, certain cookie attributes require special consideration, which we will discuss in detail.
+
+#### Path
+
+The `Path` attribute should be automatically determined by implementors. When the `SessionPersistencePolicy` policy is
+attached to an xRoute, the implementor should interpret the path as configured on the xRoute. However, if the
+`SessionPersistencePolicy` policy is attached to a service, the path should be left unset.
+
+To interpret the `Path` attribute from an xRoute, implementors should take note of the following:
+1. For an xRoute that matches all paths, the `Path` should be set to `/`.
+2. For an xRoute that has multiple paths, the `Path` should be interpreted based on the path taken.
```

Comment:
Its not clear here if 'path taken' represents the path specified in the rule that matched or the full path provided in the request?

Reply:
I think it should be the path specified in the rule that matched the full path in the request. That would then consistently apply the cookie to all paths that match the rule (which is the intent). Will let @gcs278 confirm

Reply:
That's a good question. If by "full path provided in the request", you mean the "actual path attribute provided in the incoming request", in most cases, I think you want to avoid using the actual value because it may interfere if [`PathPrefix`](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.HTTPPathMatch) is being used.

E.g. the incoming path could be `/foo/bar`, and the `PathPrefix` is `/foo`. The subsequent request for `/foo/baz` matches the `/foo` path, and would have a cookie with path `/foo/bar` which would be invalid. So I think I agree with @sanjaypujare.

Updated the language to be more specific here.